productivity-suite/piercing-library/src/piercing-fragment-host/piercing-fragment-host.ts

```diff
@@ -1,108 +1,100 @@
-import type { PiercingEvent } from "../piercing-events";
-import { piercingEventType } from "../piercing-events";
+import {
+  DispatchOptions,
+  getBus,
+  MessageBus,
+  MessageHandler,
+} from "../message-bus";
 
 // Important: we only need to import the type, do not import the class itself
 //            as it would bloat the script (which needs to be very light)
 import type { PiercingFragmentOutlet } from "../piercing-fragment-outlet";
 
-export class PiercingFragmentHost extends HTMLElement {
-	fragmentId!: string;
-	piercingEventsQueue: CustomEvent<PiercingEvent>[] = [];
-	queueEventListener?: (event: Event) => void;
-	stylesEmbeddingObserver?: MutationObserver;
+export class PiercingFragmentHost extends HTMLElement implements MessageBus {
+  private parentMessageBus!: MessageBus;
+  fragmentId!: string;
+  queueEventListener?: (event: Event) => void;
+  stylesEmbeddingObserver?: MutationObserver;
 
-	constructor() {
-		super();
-	}
+  constructor() {
+    super();
+  }
 
-	async connectedCallback() {
-		this.fragmentId = this.getAttribute("fragment-id")!;
+  private messageHandlerRemovers: (() => void)[] = [];
```

Comment:
@petebacondarwin I kept the `Remover` name, I wonder if we can find a more descriptive name? maybe `messageHandlerCleanupFunctions` or something like that? 🤔